repos/spack_repo/builtin/packages/legion/package.py

```diff
@@ -138,7 +138,8 @@ class Legion(CMakePackage, ROCmPackage):
     # cuda-centric
     cuda_arch_list = CudaPackage.cuda_arch_values
     for arch in cuda_arch_list:
-        depends_on(f"ucc cuda_arch={arch}", when=f"network=ucx +cuda cuda_arch={arch}")
+        depends_on(f"ucc cuda_arch={arch}", when=f"@25.03.0: network=ucx +cuda cuda_arch={arch}")
+        depends_on(f"ucc cuda_arch={arch}", when=f"@stable network=ucx +cuda cuda_arch={arch}")
```

Comment:
This shouldn't be required, Spack recognizes `stable` as an infinity version, ie it should be captured already with `@25.03.0:`

Reply:
ref. https://github.com/spack/spack/blob/develop/lib/spack/spack/version/common.py#L13

```
infinity_versions = ["stable", "nightly", "trunk", "head", "master", "main", "develop"]
```

Reply:
Ah, good. I didn't know it was part of that set. More cleanup incoming...